src/main/java/bio/terra/service/policy/PolicyService.java

```diff
@@ -0,0 +1,139 @@
+    logger.info("TPS base path: '{}'", policyServiceConfiguration.getBasePath());
+  }
+
+  // -- Policy Attribute Object Interface --
+  public void createSnapshotPao(UUID snapshotId, @Nullable TpsPolicyInputs policyInputs) {
+    createPao(snapshotId, TpsObjectType.SNAPSHOT, policyInputs);
+  }
+
+  public void createPao(
+      UUID resourceId, TpsObjectType resourceType, @Nullable TpsPolicyInputs policyInputs) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsPolicyInputs inputs = (policyInputs == null) ? new TpsPolicyInputs() : policyInputs;
+
+    TpsApi tpsApi = policyApi();
+    try {
+      tpsApi.createPao(
+          new TpsPaoCreateRequest()
+              .objectId(resourceId)
+              .component(TpsComponent.TDR)
+              .objectType(resourceType)
+              .attributes(inputs));
+    } catch (ApiException e) {
+      throw convertApiException(e);
+    }
+  }
+
+  public void deletePao(UUID resourceId) {
+    policyServiceConfiguration.tpsEnabledCheck();
+    TpsApi tpsApi = policyApi();
+    try {
+      try {
+        tpsApi.deletePao(resourceId);
+      } catch (ApiException e) {
+        throw convertApiException(e);
+      }
+    } catch (PolicyServiceNotFoundException e) {
+      // Not found should not cause a failure on delete.
+    }
+  }
+
+  private ApiClient getApiClient(String accessToken) {
+    ApiClient client = new ApiClient();
+    client.setAccessToken(accessToken);
+    return client;
+  }
+
+  private ApiClient getUnAuthApiClient() {
+    ApiClient client = new ApiClient();
+    client.setBasePath(policyServiceConfiguration.getBasePath());
+    return client;
+  }
+
+  private TpsApi policyApi() {
+    try {
+      return new TpsApi(
+          getApiClient(policyServiceConfiguration.getAccessToken())
+              .setBasePath(policyServiceConfiguration.getBasePath()));
+    } catch (IOException e) {
+      throw new PolicyServiceAuthorizationException(
+          "Error reading or parsing credentials file", e.getCause());
+    }
+  }
+
+  @VisibleForTesting
+  public RuntimeException convertApiException(ApiException ex) {
+    if (ex.getCode() == HttpStatus.UNAUTHORIZED.value()) {
+      return new PolicyServiceAuthorizationException(
+          "Not authorized to access Terra Policy Service", ex.getCause());
+    } else if (ex.getCode() == HttpStatus.NOT_FOUND.value()) {
+      return new PolicyServiceNotFoundException("Policy service returns not found exception", ex);
+    } else if (ex.getCode() == HttpStatus.BAD_REQUEST.value()
+        && StringUtils.containsIgnoreCase(ex.getMessage(), "duplicate")) {
+      return new PolicyServiceDuplicateException(
+          "Policy service throws duplicate object exception", ex);
+    } else if (ex.getCode() == HttpStatus.CONFLICT.value()) {
+      return new PolicyConflictException("Policy service throws conflict exception", ex);
+    } else {
+      return new PolicyServiceAPIException(ex);
+    }
+  }
```

Comment:
A few things:

- You could remove the `public` modifier and rely on the default visibility along with the `@VisibleForTesting` annotation.  That way your method is exposed for tests but not overexposed.
- This also looks like a method that should be static.
- I think you can streamline this method with a case switch expression.

```
  @VisibleForTesting
  static RuntimeException convertApiException(ApiException ex) {
    return switch (ex.getStatusCode()) {
      case HttpStatus.UNAUTHORIZED -> new PolicyServiceAuthorizationException(…);
      case HttpStatus.NOT_FOUND -> new PolicyServiceNotFoundException(…);
      …
      default -> new PolicyServiceAPIException(ex);
    };
  }
```

Reply:
I'm not sure if we can use case statements here since they need to call `.value()` to get the status code (e.g. `HttpStatus.UNAUTHORIZED.value()` instead of `HttpStatus.UNAUTHORIZED`), but IntelliJ complains that it needs to be a constant expression.

Reply:
Ah okay! The reason why I thought we could do this is that we have very similar handling in `DuosClient`, may be worth taking a look there.